pystac/item.py

```diff
@@ -431,25 +434,22 @@ def from_dict(
             raise pystac.STACTypeError(
                 f"{d} does not represent a {cls.__name__} instance"
             )
+        # These fields must exist on the dict for `matches_object_type` to pass
+        d.pop("type")
```

Comment:
I see now that these will always exist if they passed `matches_object_type`, but I think this could be more clear to someone (like me!) reading this code if we didn't mix pop and get, both with and without defaults and removed all the fields in one place.

Reply:
Yeah hopefully it feels clearer now.